react/src/page_template/settings.ts

```diff
@@ -75,19 +76,37 @@ const defaultSettings = {
     clean_background: false,
 } satisfies SettingsDictionary
 
-export function load_settings(): SettingsDictionary {
-    const settings = JSON.parse(localStorage.getItem('settings') ?? '{}') as Partial<SettingsDictionary>
-    return { ...defaultSettings, ...settings }
+export interface SettingInfo<K extends keyof SettingsDictionary> {
+    persistedValue: SettingsDictionary[K]
+    stagedValue?: SettingsDictionary[K]
 }
 
 export class Settings {
+    /**
+     * Basic Settings
+     */
     private readonly settings: SettingsDictionary
 
     constructor() {
-        this.settings = load_settings()
+        const savedSettings = localStorage.getItem('settings')
```

Comment:
I'm not sure if we want new users to just have their settings set to whatever link they were sent

Reply:
Should we just show them the staging UI? Might be a lot for a new user. Plus if they stick around they were obviously interested in what they were sent

Reply:
Yeah maybe this is fine, wasn't considering cognitive load of the staging view

Reply:
and it won't get saved unless they check a box

Reply:
I'll actually formalize this requirement in the tests

Reply:
hmm ok how about this

we save another localstorage that's a set of settings that have been user modified

don't load settings from link, enter stage mode like normal

if the set of user modified settings does not intersect with the set of settings we're staging, we hide that we're in staging mode in the UI

if the user changes one of the keys that we're staging in this hidden mode, immediately apply the stage

this avoids trivial visual modifications (e.g. opening a category) from marking a user as having settings

Reply:
Going to not do the above now ^